### DISCUSSION TITLE
Keine Verbindung über VE-Direct

### DISCUSSION THREAD

**Discussion by flurfunk0868** [hoylabs/OpenDTU-OnBattery#582] [Q&A] [answered]
Hallo zusammen und Frohes neues Jahr!
Nach langem Hin und Her, habe ich es doch noch geschafft meine Anlage umzusetzen. Leider bekomme ich anscheinend keine Werte über den VE-Direct Ausgang angezeigt. In HomeAssistant ist Open-DTU bereits über MQTT verbunden und erscheint dort auch, allerdings keine Werte. Mein Verkabelung sieht momentan so aus:
![293590846-df142d88-a8ca-41e2-bbac-695b17982a65](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/146464420/97707c3e-407e-4664-bf1c-5de7763c1fba)

Die Konsole zeigt folgendes an:
![293590892-50da4d12-dc12-4145-9a44-06a95c1702ba](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/146464420/1892c045-0d00-4c60-9ada-bdfe2c6d072b)

Weiß jemand wo das Problem liegt? Wäre für Hilfe dankbar. Schöne Grüße

Update: Ich habe ein bisschen rumprobiert und es hat sich rausgestellt, dass anscheinend ein GND Port auf meinem ESP nicht richtig funktioniert oder ich habe hier grundsätzlich was falsch verstanden. Long story short: Jetzt scheint es erstmal zu laufen, MPPT wurde erkannt aber momentan werden noch keine Werte übermittelt - liegt aber wahrscheinlich daran, dass keine Sonne scheint und der MPPT aus ist. Werde morgen nochmal schauen.

Aber vielleicht einmal was grundsätzliches: Mein ESP hat folgendes Layout und ich habe die grün eingekreisten Ports benutzt (Hat wie gesagt nicht funktioniert). Nachdem ich den GND Port gewechselt habe wie im Bild zu sehen, funktioniert jetzt alles: 
![grafik](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/146464420/6c79fca5-39f5-4a79-94b1-3608aa7fa4ca)

Frage: Sind die GND Ports immer gegenüberliegend? Habe bisher gedacht, es wäre egal welchen GND Port man nutzt....

**Comment by schlimmchen** [comment]
> Jetzt scheint es erstmal zu laufen, MPPT wurde erkannt aber momentan werden noch keine Werte übermittelt - liegt aber wahrscheinlich daran, dass keine Sonne scheint und der MPPT aus ist. Werde morgen nochmal schauen.

Woran hast du dann fest gemacht, dass VE.Direct nun funktioniert?

GND ist GND. Natürlich nicht bei sehr hohen Frequenzen weil dann plötzlich die Signallaufzeit eine Rolle spielen kann, aber das ist hier alles völlig irrelevant. Es macht Sinn, das VE.Direct nicht funktioniert, wenn GND nicht verbunden ist. Für wahrscheinlicher halte ich es aber, dass du beim Umstecken einen anderen Pin angestubst hast, oder dass der Pin vorher nicht richtig saß. Ich nehme an dass du Drahtbrücken verwendest, wenn du von "GND Port gewechselt" sprichst?

Gehts denn jetzt tatsächlich? Falls nicht, schalte "Verbose Logging" unter VE.Direct ein.

**Comment by flurfunk0868** [comment, marked as answer]
Ich habe es daran festgestellt, dass der Solarregler plötzlich in HomeAssistant auftauchte inkl. aller Entitäten/Sensoren.

Ich nutze ein Breakout-Board zur Erweiterung der Ports meines ESP und weil ich es einfach bequemer finde. Aber du hast recht, ich nutze "Jumper wire" Kabel und diese saßen alle recht locker auf dem ESP. Hat mich auch zuerst gewundert aber das Funkmodul funktionierte ja auch problemlos. Bin also erst gar nicht davon ausgegangen, dass dies die Ursache sein konnte. Dann habe ich wie gesagt den GND Pin auf einen anderen Port gelegt und alles wurde sofort erkannt. Da hab ich erstmal vermutet, dass GND nicht gleich GND ist aber jetzt weiß ich ja bescheid.
Ich habe aber nachträglich alle Kabel auf die Schraubklemmen des Breakout-Boards gelegt und seitdem hab ich weniger Sorgen wegen schlechter Kontakte.

**Comment by schlimmchen** [comment]
> Ich habe es daran festgestellt, dass der Solarregler plötzlich in HomeAssistant auftauchte inkl. aller Entitäten/Sensoren.

Die kommen auch ohne dass Tatsächlich Daten über VE.Direct kommen. Du siehst auch Daten und alles ist okay für dich?

**Reply by flurfunk0868** [reply]
Ja ich sehe die Daten, wie gesagt vorher habe ich den Regler gesehen aber keine Werte erhalten. Soweit passt alles

**Reply by schlimmchen** [reply]
Okay, wunderbar. Magst du diese Diskussion noch als "gelöst" markieren?
